src/core/server/ui_settings/ui_settings_client.ts

```diff
@@ -333,8 +333,12 @@ export class UiSettingsClient implements IUiSettingsClient {
 
     Object.entries(changes).forEach(([key, val]) => {
       if (Array.isArray(this.defaults[key]?.scope) && (this.defaults[key].scope?.length ?? 0) > 1) {
-        // if this setting has more than one scope, we should not update this setting without specifying scope
-        throw new Error(`Unable to update "${key}", because it has multiple scopes`);
+        // If this setting applies to multiple scopes, categorize it as global to maintain backward compatibility.
+        globalChanges[key] = val;
```

Comment:
Do we need to check if this multi scope setting has a global scope? And for default data source setting, we will always fallback to global scope whatever request from inside or outside workspace. Is this an expected behavior?